Sources/SwiftInspectorVisitors/Tests/TypeSyntax+NameExtractorSpec.swift

```diff
@@ -0,0 +1,111 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Foundation
+import Nimble
+import Quick
+import SwiftInspectorTestHelpers
+import SwiftSyntax
+
+@testable import SwiftInspectorVisitors
+
+final class TypeSyntaxNameExtractorSpec: QuickSpec {
+  private var sut = EnumVisitor()
+
+  override func spec() {
+    beforeEach {
+      self.sut = EnumVisitor()
+    }
+
+    describe("qualifiedName") {
+      context("when called on a TypeSyntax node representing a SimpleTypeIdentifierSyntax") {
+        final class SimpleTypeIdentifierSyntaxVisitor: SyntaxVisitor {
+          var simpleTypeIdentifier: String?
+          override func visit(_ node: SimpleTypeIdentifierSyntax) -> SyntaxVisitorContinueKind {
+            simpleTypeIdentifier = TypeSyntax(node).qualifiedName
+            return .skipChildren
+          }
+        }
+
+        var visitor: SimpleTypeIdentifierSyntaxVisitor!
+        beforeEach {
+          let content = """
+              var int: Int = 1
+              """
+
+          visitor = SimpleTypeIdentifierSyntaxVisitor()
+          try? VisitorExecutor.walkVisitor(visitor, overContent: content)
+        }
+
+        it("Finds the type") {
+          expect(visitor?.simpleTypeIdentifier) == "Int"
+        }
+      }
+
+      context("when called on a TypeSyntax node representing a MemberTypeIdentifierSyntax") {
+        final class MemberTypeIdentifierSyntaxVisitor: SyntaxVisitor {
+          var memberTypeIdentifier: String?
+          override func visit(_ node: MemberTypeIdentifierSyntax) -> SyntaxVisitorContinueKind {
+            memberTypeIdentifier = TypeSyntax(node).qualifiedName
+            return .skipChildren
+          }
+        }
+
+        var visitor: MemberTypeIdentifierSyntaxVisitor!
+        beforeEach {
+          let content = """
+              var int: Swift.Int = 1
+              """
+
+          visitor = MemberTypeIdentifierSyntaxVisitor()
+          try? VisitorExecutor.walkVisitor(visitor, overContent: content)
+        }
+
+        it("Finds the type") {
+          expect(visitor?.memberTypeIdentifier) == "Swift.Int"
+        }
+      }
+
+      context("when called on a TypeSyntax node representing an Array") {
+        final class ArraySyntaxVisitor: SyntaxVisitor {
+          override func visit(_ node: ArrayTypeSyntax) -> SyntaxVisitorContinueKind {
+            // This line should assert.
+            _ = TypeSyntax(node).qualifiedName
+            fail("This line should never be reached")
+            return .skipChildren
+          }
+        }
+
+        it("asserts") {
+          expect(try VisitorExecutor.walkVisitor(
+                  ArraySyntaxVisitor(),
+                  overContent: """
+                    var intArray: [Int] = [Int]()
+                    """))
+            .to(throwAssertion())
```

Comment:
I didn't know you could catch assertions. Nice!

Reply:
It's the one thing that makes me happy to work with Nimble tests.